libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1711,19 +1686,21 @@ IotMqttError_t IotMqtt_Wait( IotMqttOperation_t operation,
     if( _checkInit() == false )
     {
         status = IOT_MQTT_NOT_INITIALIZED;
-        goto cleanup;
     }
-
-    /* Validate the given operation reference. */
-    if( _IotMqtt_ValidateOperation( operation ) == false )
+    else
     {
-        status = IOT_MQTT_BAD_PARAMETER;
-        goto cleanup;
+        /* Validate the given operation reference. */
+        if( _IotMqtt_ValidateOperation( operation ) == false )
+        {
+            status = IOT_MQTT_BAD_PARAMETER;
+        }
+        else
+        {
+            /* Check the MQTT connection status. */
+            pMqttConnection = operation->pMqttConnection;
+        }
```

Comment:
Looks like the outermost `else` block can be removed if this `if` were changed to an `else if`